### ISSUE TITLE
I'm Landmaster, and I'm volunteering to tackle the ≥1.14.4 update.

### ISSUE THREAD

**Issue by Landmaster** [Inventory-Tweaks/inventory-tweaks#612]
Hello. I'm Landmaster, and you may know me as the creator of the Tinkers addon [PlusTiC](https://minecraft.curseforge.com/projects/plustic). I saw that you were looking for someone to bring this mod to 1.14.4 and later versions, and I'm here to ask if I can help create a rewrite of this mod in order to port it to said versions. So far, I've been able to get buttons and a basic (i.e. sorting by translation key) sorting implementation working, and [I've created a repository containing my early work on the port that I started yesterday.](https://github.com/Landmaster/InvTweaksReborn) If you guys have any questions/concerns, please feel free to reply to this issue.

BTW, thanks for the work that you've done for modded Minecraft over the years!

Kind regards,
Landmaster
*Eminentia ex minimis*

**Comment by MrLittle12** [comment]
A billion times yes!
Please.

**Comment by Kobata** [comment]
The biggest comment I'll add here is that Inventory Tweaks does a *ton* more than just a simple static sort -- the order depends on client configuration (the tree files), for the main inventory there's the whole second layer of being able to force specific items into specific slots, etc.

This means it's not really well-suited to being implemented entirely on the server, unlike, say [cpw's simpler sorter](https://github.com/cpw/inventorysorter), at least without spending a ton of effort to make sure everyone's configurations get synced to the server somehow.

(This goes kinda doubly if you end up implementing something like the API used to support the old AE2 mode where it essentially *has* to have an ordering function on the client inside AE2's list sorting)

**Comment by Landmaster** [comment]
@Kobata Fair enough. The translation key sorting that I have so far is only temporary and I'll definitely replace it with a customizable config-based one once I get the basic functionality working.

**Comment by mkalam-alami** [comment]
Synchronizing custom configs with the server seems possible, so maybe a server-based implementation can be a good approach. While very different, it would probably be more stable, performant and resistant to Minecraft API changes so I'm not against the idea.

I've never written a server mod but if I get this well, the main downside is just that the mod would only work for local games + servers where the mod is installed on both the client and server, instead of working everywhere by default. With also the concern of packet retrocompatibility between client & server I think.

Thanks for tackling this and best wishes for the rewrite 😄

(PS. When you get to custom configs, I hope using the same file names and format will be considered! Making IT Reborn a drop-in replacement without having players rewrite the config would be a killer feature)

**Comment by Landmaster** [comment]
I've already pushed a few beta builds for the rewrite, though I'm waiting for CurseForge to approve the project.

**Comment by Landmaster** [comment]
First builds are released at https://www.curseforge.com/minecraft/mc-mods/it-reborn!

**Comment by Sheepotastic** [comment]
Like the idea, don't like that its stuck with needing to be installed client side, makes it kinda useless for me, I don't play singleplayer or on servers where I can freely add mods to them

**Comment by Announcement** [comment]
Anything I can do to help? give me instructions api links for tasks that need to be written/converted, documentation or test suites to write and i got you :p I'm just not really up for handling and investigating on my own at this moment without some encouragement/motivation/interest

**Comment by TheBeardedQuack** [comment]
Hi Landmaster,
I would also like to volunteer some of my time to aid with rebuilding this mod. While I have barely any Java experience (I did a few tiny things with the old mod loader way back when) I have worked with C++ and C# professionally for the last 5 years. I can help test and track down bugs with the stack traces, even if I can't fix bugs I can help provide detailed steps to reproduce and find.

I was looking for small open source mod to get help contribute to in order to expand my expertise with Java and I loved using inventory tweaks, I'd like to help bring it up to date.

**Comment by Landmaster** [comment]
@TheBeardedQuack Check out the project's GitHub at https://github.com/Landmaster/InvTweaksReborn

Also, Inventory Tweaks is now on 1.15.2!

**Comment by FinchHub** [comment]
Ah, man. The entire charm of InvTweaks was that it could be used client-side; I was looking forward to the idea of being able to use this on a realm. That sucks. 

At least it's being worked on again. Good effort.

**Comment by Landmaster** [comment]
@Vulano The client-side functionality is on my to-do list (https://github.com/Landmaster/InvTweaksReborn/issues/8).

**Comment by Lyraedan** [comment]
I'd be more then happy to rewrite InventoryTweaks from the ground up for compatibility with Fabric.
It'll function exactly the same to the end user they wouldn't even notice a difference.
I'll do this for 1.15+
